### PR TITLE
Update kvutils Config with missing flags options from SandboxConfig [DPP-864]

### PR DESCRIPTION
**Pull request by tudor-da**
Update kvutils Config and SandboxConfig
* Remove `eager-package-loading` and `scenario` from SandboxConfig
* Add `profile-dir` and `stackTraces` to kvutils Config
* Configurable authService in kvutils

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by SamirTalwar-DA** [comment]
How come? Do we need these for kvutils?

**Comment by tudor-da** [comment]
> How come? Do we need these for kvutils?

Not that we do explicitly need them, but for alignment. Let's keep this topic open, but @mziolekda came with a good proposal to have a single config for the sandboxes so we can decided to enrich the `kvutils.app.Config` with what we found as missing from `SandboxConfig`, hence these changes

**Comment by SamirTalwar-DA** [comment]
Not necessarily, just wanted to make sure they were actually useful. Thanks for doing this!

**Comment by nicu-da** [comment]
> > How come? Do we need these for kvutils?
> 
> Not that we do explicitly need them, but for alignment. Happy to keep this topic open, but @mziolekda came with a good proposal to have a single config for the sandboxes so we decided to propose enrichment of the `kvutils.app.Config` with what we found as missing from `SandboxConfig`, hence these changes. Do you see downsides to adding these options?

Do we plan in using a single scala model for the config?  How will that work out in the future if we move kvutils from the sdk?

**Comment by tudor-da** [comment]
> > > How come? Do we need these for kvutils?
> > 
> > 
> > Not that we do explicitly need them, but for alignment. Happy to keep this topic open, but @mziolekda came with a good proposal to have a single config for the sandboxes so we decided to propose enrichment of the `kvutils.app.Config` with what we found as missing from `SandboxConfig`, hence these changes. Do you see downsides to adding these options?
> 
> Do we plan in using a single scala model for the config? How will that work out in the future if we move kvutils from the sdk?

I think the best way to go forward is to have a single config structure and CLI, allowing seamless portability between sandbox-on-x (which is becoming the wide-spread test-bed for the SDK and performance testing in LR) and PKV. The kvutils Config is best suited for this since it also allows configuration of ledger-side parameters (via the `Extra` config). I'd suggest, then, to keep the kvutils Config API in the SDK and internalize it (maybe in `sandbox-common`?). What do you think?

**Comment by SamirTalwar-DA** [comment]
I think these configuration files will diverge drastically after splitting KV from this repository. I expect we will just duplicate it.

In the long term, we expect to replace KV configuration parsing with something like [HOCON](https://github.com/lightbend/config).

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala [hunk 1]
```diff
@@ -96,6 +103,10 @@ object Config {
       extra = extra,
```
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:103]: Please sort.

File: ledger/sandbox-on-x/src/classic/scala/com/daml/platform/ledger/ConfigConverter.scala [hunk 2]
```diff
@@ -98,6 +92,10 @@ object ConfigConverter {
       extra = extraBridgeConfig,
```
Comment (SamirTalwar-DA) [inline ledger/sandbox-on-x/src/classic/scala/com/daml/platform/ledger/ConfigConverter.scala:92]: Sort.

**Review by SamirTalwar-DA — APPROVED** [review]

**Comment by SamirTalwar-DA** [comment]
Changed my mind.

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala [hunk 3]
```diff
@@ -663,6 +674,26 @@ object Config {
           .action((value, config: Config[Extra]) =>
             config.withUserManagementConfig(_.copy(maxUsersPageSize = value))
           )
+        opt[Unit]('s', "static-time")
```
[the remark this answers is not in the document]
Comment (tudor-da) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:677]: Done

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala (hunk 1 above)
Comment (tudor-da) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala]: Sorted

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala [hunk 4]
```diff
@@ -19,39 +16,49 @@ import com.daml.metrics.MetricsReporter
...
     acsGlobalParallelism: Int,
+    acsIdFetchingParallelism: Int,
+    acsIdPageSize: Int,
     acsIdQueueLimit: Int,
-    stateValueCache: caching.WeightedCache.Configuration,
-    lfValueTranslationEventCache: caching.SizedCache.Configuration,
+    configurationLoadTimeout: Duration,
+    commandConfig: CommandConfiguration,
+    enableInMemoryFanOutForLedgerApi: Boolean,
+    enableSelfServiceErrorCodes: Boolean,
```
Comment (mziolekda) [inline ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:43]: You have a race on that parameter with @kamil-da's PR

Comment (tudor-da) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:43]: Someone will have to rebase. I'm fine if it's going to be me

File: ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/BridgeConfig.scala [hunk 5]
```diff
@@ -47,51 +37,9 @@ object BridgeConfigProvider extends ConfigProvider[BridgeConfig] {
...
-          "Enabling stack traces may have a significant performance impact."
-      )
-
-    JwtVerifierConfigurationCli.parse(parser)((v, c) =>
-      c.copy(extra = c.extra.copy(authService = AuthServiceJWT(v)))
-    )
-
-    parser.checkConfig(c =>
-      Either.cond(
-        c.maxDeduplicationDuration.forall(_.compareTo(Duration.ofHours(1L)) <= 0),
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/BridgeConfig.scala:81]: Intentional removal?

Comment (tudor-da) [reply ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/BridgeConfig.scala:81]: Unfortunately yes. Too many rebase attempts.

**Pull request merged**